Common++/Makefile

```diff
@@ -40,7 +40,7 @@ endif
 			
 Obj/%.o: src/%.cpp Obj/%.d
 	@echo Building file: $<
-	@$(CXX) $(DEPS) $(INCLUDES) $(PCAPPP_BUILD_FLAGS) -c -fmessage-length=0 -MMD -MP -MF "$(@:Obj/%.o=Obj/%.d)" -MT "$@" -o "$@" "$<"
+	@$(CXX) $(DEPS) $(INCLUDES) $(PCAPPP_INCLUDES) $(PCAPPP_BUILD_FLAGS) -c -fmessage-length=0 -MMD -MP -MF "$(@:Obj/%.o=Obj/%.d)" -MT "$@" -o "$@" "$<"

```

Comment:
This change is not needed, `Common++` - `$(INCLUDES)` already has all the includes needed to compile this library

Reply:
I agree, it should not be, but --cflags-only-other is for me "-include rte_config.h -march=corei7" and we included it in $(PCAPPP_BUILD_FLAGS) in commit b9f00a9c85b312e9fe84a8b4093edc3985dd425d

Reply:
I'm not sure I understand... 🤔

Reply:
$ pkg-config --cflags-only-other libdpdk
-include rte_config.h -march=corei7

in mk/PcapPlusPlus.mk.dpdk :
TMPFLAG = $(shell $(PKGCONF) --cflags-only-other libdpdk 2>/dev/null)
ifneq ($(TMPFLAG),)
PCAPPP_BUILD_FLAGS += $(TMPFLAG)
endif

Reply:
I'm not sure how this is related to the `Makefile` in `Common++`...
Basically `Common++` and `Packet++` libraries don't use any DPDK object, so they shouldn't need any additional DPDK include. Only the `Pcap++` library needs DPDK header files, which is why the change you made is necessary

Reply:
Common++/Makefile includes mk/PcapPlusPlus.mk which in turn has mk/PcapPlusPlus.mk.dpdk included, so PCAPPP_BUILD_FLAGS ends up including the DPDK stuff. $(INCLUDES) $(PCAPPP_BUILD_FLAGS) seems asymmetric to me, maybe there should be a BUILD_FLAGS too?
Or should the PcapPlusPlus.mk.dpdk restricted to Pcap++ ?

Reply:
`PCAPPP_INCLUDES`, `PCAPPP_LIBS`, `PCAPPP_LIBS_DIR` and  `PCAPPP_BUILD_FLAGS` are mostly for external users who want to use PcapPlusPlus.
PcapPlusPlus own libraries (`Common++`, `Packet++`, `Pcap++`) have their own includes so they don't need to user `PCAPPP_*`